relay/chunkstore/chunk_reader.go

```diff
@@ -0,0 +1,110 @@
+package chunkstore
+
+import (
+	"context"
+	"fmt"
+	"github.com/Layr-Labs/eigenda/common/aws/s3"
+	"github.com/Layr-Labs/eigenda/disperser"
+	"github.com/Layr-Labs/eigenda/disperser/common/blobstore"
+	"github.com/Layr-Labs/eigenda/encoding"
+	"github.com/Layr-Labs/eigenda/encoding/rs"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/consensys/gnark-crypto/ecc/bn254"
+)
+
+// ChunkReader reads chunks written by ChunkWriter.
+type ChunkReader interface {
+	// GetChunkProofs reads a slice of proofs from the chunk store.
+	GetChunkProofs(ctx context.Context, blobKey disperser.BlobKey) ([]*encoding.Proof, error)
+	// GetChunkCoefficients reads a slice of frames from the chunk store. The metadata parameter
+	// should match the metadata returned by PutChunkCoefficients.
+	GetChunkCoefficients(
+		ctx context.Context,
+		blobKey disperser.BlobKey) ([]*rs.Frame, error)
+}
+
+var _ ChunkReader = (*chunkReader)(nil)
+
+type chunkReader struct {
+	logger        logging.Logger
+	metadataStore *blobstore.BlobMetadataStore
+	client        s3.Client
+	bucket        string
+	shards        []uint32
+}
+
+// NewChunkReader creates a new ChunkReader.
+//
+// This chunk reader will only return data for the shards specified in the shards parameter.
+// If empty, it will return data for all shards. (Note: shard feature is not yet implemented.)
+func NewChunkReader(
+	logger logging.Logger,
+	metadataStore *blobstore.BlobMetadataStore,
+	s3Client s3.Client,
+	bucketName string,
+	shards []uint32) ChunkReader {
+
+	return &chunkReader{
+		logger:        logger,
+		metadataStore: metadataStore,
+		client:        s3Client,
+		bucket:        bucketName,
+		shards:        shards,
+	}
+}
+
+func (r *chunkReader) GetChunkProofs(
+	ctx context.Context,
+	blobKey disperser.BlobKey) ([]*encoding.Proof, error) {
```

Comment:
Seems like this blobkey references the v1 blob key. In StoreBlob for V2 we use blobKey.Hex() = string https://github.com/Layr-Labs/eigenda/blob/c63dd61bf633c82ba6b30e7f71c5a17e5c452b7c/disperser/common/v2/blobstore/s3_blob_store.go#L27. 

V2 blob key:
```
type BlobKey [32]byte

func (b BlobKey) Hex() string {
	return hex.EncodeToString(b[:])
}
```

Reply:
Also when we fetch for proofs vs coefficients don't we need a different S3 key to differentiate it?

Reply:
I've been assuming we'd use different buckets. Started a slack conversation to discuss. Will circle back on this once we decide how we want to handle buckets and name spacing.

Reply:
I've switched over to using `v2.BlobKey` as recommended by @ian-shim.